src/main/java/com/sngular/kloadgen/loadgen/impl/JsonSRLoadGenerator.java

```diff
@@ -48,18 +49,16 @@ public void setUpGenerator(final Map<String, String> originals, final String avr
 
   public void setUpGenerator(final String schema, final List<FieldValueMapping> fieldExprMappings) {
     final var parsedSchema = new JsonSchemaProvider().parseSchema(schema, Collections.emptyList(), true);
-    metadata = parsedSchema.map(parsSchema -> Pair.of(new SchemaMetadata(1, 1, "JSON", Collections.emptyList(), schema), parsSchema)).orElse(null);
+    metadata =
+        parsedSchema.map(parsSchema -> Pair.of(new BaseSchemaMetadata(ConfluentSchemaMetadata.parse(new SchemaMetadata(1, 1, "JSON", Collections.emptyList(), schema))),
```

Comment:
If in line 55, we have `SchemaTypeEnum.JSON`, that has the vale `"JSON"`, I think it's better to use this constant here.

Reply:
Done!